tests/validators/test_is_subclass.py

```diff
@@ -57,9 +57,11 @@ def test_not_parent():
     assert not v.isinstance_python(Foo)
 
 
-def test_invalid_type():
-    with pytest.raises(SchemaError, match="TypeError: 'Foo' object cannot be converted to 'PyType"):
-        SchemaValidator(core_schema.is_subclass_schema(Foo()))
+@pytest.mark.parametrize('input_value', [Foo, Foobar, Bar, type])
+def test_is_subclass_object_of_class(input_value):
+    v = SchemaValidator(core_schema.is_subclass_schema(Foo()))
+    with pytest.raises(TypeError, match='arg 2 must be a class'):
+        v.isinstance_python(input_value)
```

Comment:
Does this fix the actual issue? I.e. does `type[int | str]` work as a field annotation with this? Or is it just making it handle some more cases (things that aren't a PyType) but not the type of a union case?

Reply:
You're right, I think the issues is with the `_subclass_schem` a function, because in the case of Union we have to use the same condition as https://github.com/pydantic/pydantic/blob/66038bee3a3598eefd011d3bea19d7296e3dd7bc/pydantic/_internal/_generate_schema.py#L1198 and return an `is_instance_schema` and not a subclass schema. WDYT?